src/storage-client/src/types/connections.rs

```diff
@@ -589,14 +589,10 @@ impl CsrConnection {
                 // computed for every request. This ensures that, if the tunnel
                 // fails and restarts at a new address, requests will start
                 // using the new tunnel address.
-                client_config = client_config.add_proxy(mz_ccsr::Proxy::custom(move |url| {
+                client_config = client_config.set_tunnel_override_callback(move || {
                     let addr = ssh_tunnel.local_addr();
-                    let mut url = url.clone();
-                    url.set_host(Some(&addr.ip().to_string()))
-                        .expect("cannot fail");
-                    url.set_port(Some(addr.port())).expect("cannot fail");
-                    Some(url)
-                }));
+                    Url::parse(&format!("http://{}:{}", addr.ip(), addr.port())).unwrap()
```

Comment:
@benesch does this need to be the `scheme()` of the existing url, instead of always `http`?

Reply:
we'll have to pass the existing url in, but not too bad

Reply:
oh wait nvm!